### PR TITLE
Unify the logic on sidecar injection and service creation

### PR DESCRIPTION
**Pull request by JasonChen86899**
# Description

#6967

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by JasonChen86899** [comment]
@yaron2

File: pkg/injector/service/injector.go [hunk 1]
```diff
@@ -45,14 +45,13 @@ const (
 
 var log = logger.NewLogger("dapr.injector.service")
 
+// AllowedServiceAccountInfos default allowed service accounts
+// You can add other accounts which you want by adding them in helm chart file 'dapr_sidecar_injector/values.yaml'
+// helm chart value: allowedServiceAccounts or allowedServiceAccountsPrefixNames
 var AllowedServiceAccountInfos = []string{
 	"kube-system:replicaset-controller",
 	"kube-system:deployment-controller",
-	"kube-system:cronjob-controller",
```
Comment (daixiang0) [inline pkg/injector/service/injector.go:51]: Those service accounts are default for many versions, why remove them?

Comment (JasonChen86899) [reply pkg/injector/service/injector.go:51]: > Those service accounts are default for many versions, why remove them?

I have moved them to helm chart default values `allowedServiceAccounts` and not affect  pervious versions.

You can see the detail reason and design for this PR in this issue #6967

**Comment by JasonChen86899** [comment]
add unit test for new code

**Comment by daixiang0** [comment]
Please fix conflicts.

File: cmd/operator/options/options.go [hunk 2]
```diff
@@ -109,5 +115,30 @@ func New() *Options {
...
+			resourceSet[&batchv1.Job{}] = struct{}{}
+		case "DaemonSet":
+			resourceSet[&appsv1.DaemonSet{}] = struct{}{}
+		default:
+			log.Errorf("unSupported k8s resource [%v] for service creation", resource)
+		}
+	}
+	resources := make([]client.Object, 0, len(resourceSet))
+	for resource := range resourceSet {
+		resources = append(resources, resource)
```
Comment (daixiang0) [inline cmd/operator/options/options.go:139]: Why not append `opts.AllowedServiceResources` directly?

Comment (JasonChen86899) [reply cmd/operator/options/options.go:139]: following is the reason for my logic
1. remove duplicate user-inputs
2. error log for unSupported k8s resource

**Comment by yaron2** [comment]
@JasonChen86899 please resolve the conflicts

**Comment by JasonChen86899** [comment]
> @JasonChen86899 please resolve the conflicts

done

**Comment by JasonChen86899** [comment]
@yaron2 this pr has closed for a long time, do it can be opened?  I have already resolved the conflicts and merge latest code

**Pull request closed without merging**